pkg/receiver/gitproviderreceiver/internal/scraper/githubscraper/graphql_helpers.go

```diff
@@ -22,14 +22,63 @@ func getRepoData(
 	repoCursor *string,
 	// since we're using a interface{} here we do type checking when data
 	// is returned to the calling function
-) (interface{}, error) {
+) (*getRepoDataBySearchResponse, error) {
 	data, err := getRepoDataBySearch(ctx, client, searchQuery, repoCursor)
 	if err != nil {
 		return nil, err
 	}
 	return data, nil
 }
 
+type mockClient struct {
```

Comment:
Why are providing `mockClient` to the `MakeRequest` function? This is for testing only correct? Should go in `helpers_test.go`

Reply:
This is only for testing yeah.  I thought helpers_test.go was just the same thing but on the gitlab side no?